client/boot/index.js

```diff
@@ -161,8 +160,39 @@ function boot() {
 
 	translatorJumpstart.init();
 
-	reduxStore = createReduxStore();
+	localforage.config( {
+		name: 'calypso',
+		storeName: 'calypso_store',
+		description: 'Calypso Storage'
+	} );
+	localforage.getItem( 'redux-state', ( error, initialState ) => {
+		let reduxStore;
+		if ( error ) {
+			debug( 'failed to load initial redux-store state', error );
+			reduxStore = createReduxStore();
+		} else {
+			debug( 'loading redux-store with initial state', initialState );
+			reduxStore = createReduxStore( initialState || {} );
+		}
```

Comment:
I see, `null` doesn't trigger the default parameter. Next :question: are there reasonable expectations that we would send `null`, since it appears like we would have to _explicitly_ save `null` for the state, which doesn't seem too likely to me